### ISSUE TITLE
ES6 backticks not interpreted correctly

### ISSUE THREAD

**Issue by IBwWG** [fdorg/flashdevelop#1427]
For example, save a file as .js containing:

```es6
const a = `b = ${b}`;
const b = `http://${host}/mydir/`;
'string';
// comment
```

In the case of `a`, the `b =` is black as if it were code.  Only the `${b}` is code, and `b =` is like a string.
With `b`, the `//` is interpreted as a comment (which it isn't) and the rest of the line is green.

Compare github's rendering of the above with FlashDevelop's.

This is on 5.1.0.3, win7x64sp1 (no updates available when I checked just now.)

**Comment by Neverbirth** [comment]
This case is styled by Scintilla with a key not defined by FlashDevelop. Although I wouldn't be surprised if the version used right now would have problems with it, it will be updated soon.

You could go to your Settings/Languages folder, edit the JScript.xml file, and under `<use-styles>` add `<style name="stringraw" fore="0xa31515" />`.

**Comment by IBwWG** [comment]
Thanks...that didn't seem to change anything (at least for case `b`) though...

**Comment by Neverbirth** [comment]
Oh, it's not enough with adding this style, an extra property should be defined, so with current build you wouldn't be able to make it working even if that version of Scintilla supports it (I ignore if it's the case)... my build with updated Scintilla allows for it tho.

![image](https://cloud.githubusercontent.com/assets/2273861/21676466/f7f74f66-d335-11e6-927c-d626c2b74f24.png)

**Comment by Neverbirth** [comment]
![image](https://cloud.githubusercontent.com/assets/2273861/21759987/ebbd02f2-d648-11e6-8230-46f256ba1f11.png)

String interpolation works now too in the internal Scintilla build used here. This implementation is generic and would be able to support multiple interpolation syntaxes, so hopefully it will be approved by the Scintilla people. Still a couple of things missing in Haxe, but for most use cases it should be fine:

![image](https://cloud.githubusercontent.com/assets/2273861/21760312/68f42e10-d64b-11e6-83a4-801b95d96a9c.png)

**Comment by NemoStein** [comment]
I wrote a [post](http://www.flashdevelop.org/community/viewtopic.php?f=13&t=12658) in the forums six months ago and bumped it just now, then I thought "Oh, I should check the github instead!".

As I could understand, the current Scintilla is able to render string interpolation as raw string and your, modified, version is able to handle the interpolation just fine, correct?

That being the case, could we receive this new version in some foreseeable future in FD?

**Comment by Neverbirth** [comment]
That's right, this change will hopefully add string interpolation support in a way Scintilla developers will be able to add it also to other languages.

I cannot give an exact date, but I don't think it should take too much longer.
